scripts/setup-helper-functions.sh

```diff
@@ -58,81 +58,59 @@ function github_checkout {
-# Sets and exports the variable VELOX_CXX_FLAGS with appropriate compiler flags.
-# If $CPU_ARCH is set then we use that else we determine best possible set of flags
-# to use based on current cpu architecture.
+# get_cxx_flags
 # The goal of this function is to consolidate all architecture specific flags to one
 # location.
+# Sets and exports the variable VELOX_CXX_FLAGS with appropriate compiler flags.
+# We determine best possible set of flags to use based on the cpu architecture CPU_ARCH.
 # The values that CPU_ARCH can take are as follows:
 #   arm64  : Target Apple silicon.
 #   aarch64: Target general 64 bit arm cpus.
 #   avx:     Target Intel CPUs with AVX.
 #   sse:     Target Intel CPUs with sse.
 # Echo's the appropriate compiler flags which can be captured as so
-# CXX_FLAGS=$(get_cxx_flags) or
-# CXX_FLAGS=$(get_cxx_flags "avx")
+# CXX_FLAGS=$(get_cxx_flags)
 
 function get_cxx_flags {
-  local CPU_ARCH=$1
-
-  local OS
-  OS=$(uname)
-  local MACHINE
-  MACHINE=$(uname -m)
-  ADDITIONAL_FLAGS=""
+  local CPU_ARCH="avx"
+  local OS=$(uname)
+  local MACHINE=$(uname -m)
 
-  if [[ -z "$CPU_ARCH" ]] || [[ $CPU_ARCH == "unknown" ]]; then
-    if [ "$OS" = "Darwin" ]; then
-
-      if [ "$MACHINE" = "x86_64" ]; then
-        local CPU_CAPABILITIES
-        CPU_CAPABILITIES=$(sysctl -a | grep machdep.cpu.features | awk '{print tolower($0)}')
-
-        if [[ $CPU_CAPABILITIES =~ "avx" ]]; then
+  if [ "$OS" = "Darwin" ]; then
+    if [ "$MACHINE" = "arm64" ]; then
+      CPU_ARCH="arm64"
+    fi
+  elif [ "$OS" = "Linux" ]; then
+    if [ "$MACHINE" = "aarch64" ]; then
+          CPU_ARCH="aarch64"
+    else
+      local CPU_CAPABILITIES=$(cat /proc/cpuinfo | grep flags | head -n 1| awk '{print tolower($0)}')
+      # Even though the default is avx, we need this check since avx machines support sse as well.
+      if [[ $CPU_CAPABILITIES =~ "avx" ]]; then
           CPU_ARCH="avx"
-        else
+      elif [[ $CPU_CAPABILITIES =~ "sse" ]]; then
           CPU_ARCH="sse"
-        fi
-
-      elif [[ $(sysctl -a | grep machdep.cpu.brand_string) =~ "Apple" ]]; then
-        # Apple silicon.
-        CPU_ARCH="arm64"
-      fi
-
-    # On MacOs prevent the flood of translation visibility settings warnings.
-    ADDITIONAL_FLAGS="-fvisibility=hidden -fvisibility-inlines-hidden"
-    else [ "$OS" = "Linux" ];
-
-      local CPU_CAPABILITIES
-      CPU_CAPABILITIES=$(cat /proc/cpuinfo | grep flags | head -n 1| awk '{print tolower($0)}')
-
-      if [[ "$CPU_CAPABILITIES" =~ "avx" ]]; then
-            CPU_ARCH="avx"
-      elif [[ "$CPU_CAPABILITIES" =~ "sse" ]]; then
-            CPU_ARCH="sse"
-      elif [ "$MACHINE" = "aarch64" ]; then
-            CPU_ARCH="aarch64"
       fi
     fi
+  else
+    echo "Unsupported platform $OS"; exit 1; 
   fi
-
   case $CPU_ARCH in
 
     "arm64")
-      echo -n "-mcpu=apple-m1+crc -std=c++17 -fvisibility=hidden $ADDITIONAL_FLAGS"
+      echo -n "-mcpu=apple-m1+crc -std=c++17 -fvisibility=hidden"
```

Comment:
Do we not need `-fvisibility-inlines-hidden` being added here? It was previously present to avoid log flooding.

Reply:
I don't see the log messages after removing that. We need to evaluate and specify `-fvisibility=hidden` and `-fvisibility-inlines-hidden` for both clang and gcc.